chia/daemon/client.py

```diff
@@ -45,23 +45,27 @@ async def start(self):
             await self.close()
             raise
 
-        async def listener():
+        async def listener_task() -> None:
+            await self.listener()
+            await self.close()
+
+        asyncio.create_task(listener_task())
+        await asyncio.sleep(1)
+
+    async def listener(self) -> None:
+        if self.websocket is not None:
```

Comment:
It would never be none, but mypy needed it

Reply:
Then we would _really_ want to know that something impossible has happened...  Also, when you are placating mypy and you don't have a link to a mypy issue or pr that would eliminate the need to placate mypy, then it is usually an indication that there's a design issue in our code.  Albeit maybe pre-existing here.

Reply:
Also, instead of ending up indenting everything for an impossible case, handle the impossible case on the side most likely by raising an exception.

Also, also, don't assume stuff controlled outside that method is impossible to get into a state that is in fact sometimes the state of it.  We want code that helps you when you break expectations that have been made.  Not code that silently does nothing when you modify it and end up violating an assumption.